manticore/platforms/linux.py

```diff
@@ -1225,15 +1247,13 @@ def sys_fsync(self, fd):
 
         ret = 0
         try:
-            f = self.files[fd]
-            if isinstance(f, Socket):
-                ret = -errno.EINVAL
-            else:
-                f.sync()
+            self.files[fd].sync()
         except IndexError:
             ret = -errno.EBADF
+        except BadFd:
```

Comment:
it's confusing that BadFd returns EINVAL but right above IndexError returns EBADF? i would expect BadFd to return EBADF